ui/src/app/pages/analysis/notebook-redirect.spec.tsx

```diff
@@ -0,0 +1,154 @@
+    authenticating: 1,
+    creatingCopying: 2,
+    redirecting: 3
+  };
+
+  const mountedComponent = () => {
+    return mount(<NotebookRedirect/>);
+  };
+
+  const getCardSpinnerTestId = (cardId) => {
+    return '[data-test-id="progress-card-spinner-' + cardId + '"]';
+  };
+
+  beforeEach(() => {
+    clusterStub = new ClusterApiStub();
+    clusterStub.cluster.status = ClusterStatus.Creating;
+
+    registerApiClient(ClusterApi, clusterStub);
+    registerApiClientNotebooks(JupyterApi, new JupyterApiStub());
+    registerApiClientNotebooks(NotebooksApi, new NotebooksApiStub());
+    registerApiClientNotebooks(NotebooksClusterApi, new NotebooksClusterApiStub());
+
+    serverConfigStore.next({gsuiteDomain: 'x'});
+    urlParamsStore.next({
+      ns: WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
+      wsid: WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
+      nbName: 'blah blah'
+    });
+    queryParamsStore.next({
+      kernelType: Kernels.R,
+      creating: true
+    });
+    currentWorkspaceStore.next(workspace);
+    userProfileStore.next({profile, reload, updateCache});
+
+    // mock timers
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.useRealTimers();
+    jest.clearAllMocks();
+  });
+
+  it('should render', () => {
+    const wrapper = mountedComponent();
+    expect(wrapper).toBeTruthy();
+    wrapper.unmount();
+  });
+
+  it('should show redirect display before showing notebook', async() => {
+    const wrapper = mountedComponent();
+    expect(wrapper.exists('[data-test-id="notebook-redirect"]')).toBeTruthy();
+    wrapper.unmount();
+  });
+
+  it('should be "Initializing" until a Creating cluster for an existing notebook is running', async() => {
+    const wrapper = mountedComponent();
+
+    wrapper.setState({creatingNewNotebook: false});
+    clusterStub.cluster.status = ClusterStatus.Creating;
+
+    await expectRedirectingAfterRunning(wrapper);
+
+    wrapper.unmount();
+  });
+
+  it('should be "Initializing" until a Creating cluster for a new notebook is running', async() => {
+    const wrapper = mountedComponent();
+
+    wrapper.setState({creatingNewNotebook: true});
+    clusterStub.cluster.status = ClusterStatus.Creating;
+
+    await expectRedirectingAfterRunning(wrapper);
+
+    wrapper.unmount();
+  });
+
+  it('should be "Resuming" until a Stopped cluster for an existing notebook is running', async() => {
+    const wrapper = mountedComponent();
+
+    wrapper.setState({creatingNewNotebook: false});
+    clusterStub.cluster.status = ClusterStatus.Stopped;
+
+    await expectRedirectingAfterRunning(wrapper);
+
+    wrapper.unmount();
+  });
+
+  it('should be "Resuming" until a Stopped cluster for a new notebook is running', async() => {
+    const wrapper = mountedComponent();
+
+    wrapper.setState({creatingNewNotebook: true});
+    clusterStub.cluster.status = ClusterStatus.Stopped;
+
+    await expectRedirectingAfterRunning(wrapper);
+
+    wrapper.unmount();
```

Comment:
Interesting - I would not have thought this would be necessary. Does unmount not happen automatically at the end of these test cases?

Reply:
It does not seem to be necessary - removed